src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -72,25 +76,19 @@ private static LineMapData toMapData(Line line) {
             .terminal1Connected(terminal1.isConnected())
             .terminal2Connected(terminal2.isConnected())
             .voltageLevelId1(terminal1.getVoltageLevel().getId())
-            .voltageLevelId2(terminal2.getVoltageLevel().getId());
-        if (!Double.isNaN(terminal1.getP())) {
-            builder.p1(terminal1.getP());
-        }
-        if (!Double.isNaN(terminal1.getQ())) {
-            builder.q1(terminal1.getQ());
-        }
-        if (!Double.isNaN(terminal2.getP())) {
-            builder.p2(terminal2.getP());
-        }
-        if (!Double.isNaN(terminal2.getQ())) {
-            builder.q2(terminal2.getQ());
-        }
-        if (!Double.isNaN(terminal1.getI())) {
-            builder.i1(terminal1.getI());
-        }
-        if (!Double.isNaN(terminal2.getI())) {
-            builder.i2(terminal2.getI());
-        }
+            .voltageLevelId2(terminal2.getVoltageLevel().getId())
+            .p1(!Double.isNaN(terminal1.getP()) ? terminal1.getP() : null)
```

Comment:
Why not simplify and let null be on the '?' side ?

Reply:
done